azure/Kqlmagic/kusto_client.py

```diff
@@ -80,19 +94,47 @@ def __init__(self, conn_kv:dict):
         authority : 'microsoft.com', optional
             In case your tenant is not microsoft please use this param.
         """
+        cloud = options.get("cloud")
 
         cluster_name = conn_kv[ConnStrKeys.CLUSTER]
-        data_source = cluster_name if cluster_name.find("://") >= 0 else self._DATA_SOURCE_TEMPLATE.format(cluster_name)
+
+        if cluster_name.find("://") >= 0:
+            data_source = cluster_name
+            cloud = cloud or self.getCloudFromHTTP(data_source)
+
+        else:
+            cloud = cloud or "public"
+            if cloud.find("://") >=0:
+                raise Exception("error: use url for cluster name")
+            data_source = self._DATA_SOURCE_TEMPLATE.format(cluster_name, self._CLOUD_URLS.get(cloud))
 
         self._mgmt_endpoint = self._MGMT_ENDPOINT_TEMPLATE.format(data_source, self._MGMT_ENDPOINT_VERSION)
         self._query_endpoint = self._QUERY_ENDPOINT_TEMPLATE.format(data_source, self._QUERY_ENDPOINT_VERSION)
-        _FQN_DRAFT_PROXY_CLUSTER_PATTERN = re.compile(r"http(s?)\:\/\/ade\.(int\.)?(applicationinsights|loganalytics)\.io.*$")
+        _FQN_DRAFT_PROXY_CLUSTER_PATTERN = re.compile(r"http(s?)\:\/\/ade\.(int\.)?(applicationinsights|loganalytics)\.(io|cn|us|de).*$")
         auth_resource = data_source
 
         if _FQN_DRAFT_PROXY_CLUSTER_PATTERN.match(data_source):
-            auth_resource = "https://kusto.kusto.windows.net"
-
-        self._aad_helper = _MyAadHelper(ConnKeysKCSB(conn_kv, auth_resource), self._DEFAULT_CLIENTID) if conn_kv.get(ConnStrKeys.ANONYMOUS) is None else None
+            if cloud.find("://") >=0:
+                raise Exception("error: url for cloud not supported")
+            auth_resource = "https://kusto.kusto.{0}".format(self._CLOUD_URLS.get(cloud))
+
+        self._aad_helper = _MyAadHelper(ConnKeysKCSB(conn_kv, auth_resource), self._DEFAULT_CLIENTID, **options) if conn_kv.get(ConnStrKeys.ANONYMOUS) is None else None
+
+
+    def getCloudFromHTTP(self, http):
```

Comment:
this function is currently not needed. But don't remove it, as we can later to add one more cloud option: 'auto' which will mean that the cloud will be derived from the url (in this case if a cluster is specified which is not a full url, we will assume public)